cmd/metrics.go

```diff
@@ -35,16 +35,33 @@ func isValidMetric(metric string) bool {
 	return false
 }
 
-func getMetricDependenciesDirect(adjacencyList AdjacencyList) MetricDependenciesDirectType {
-	depsCount := make(MetricDependenciesDirectType)
+/*
+Get direct dependencies given adjacency list (used both for dependencies and dependents)
+as it's just a mapping of node -> [nodes].
+*/
+func getMetricDependenciesDirect(adjacencyList AdjacencyList) GenericMapStringToAny {
+	depsCount := make(GenericMapStringToAny)
 	for key, deps := range adjacencyList {
 		depsCount[key] = len(deps)
 	}
+	// set count to 0 for nodes that were not present as keys in the adjacency list
+	// (i.e. they are only dependencies of some nodes and do not depend on anything)
+	for _, deps := range adjacencyList {
+		for _, dep := range deps {
+			if _, isKey := adjacencyList[dep]; !isKey {
+				depsCount[dep] = 0
+			}
+		}
+	}
 	return depsCount
 }
 
-func getMetricDependenciesTransitive(adjacencyList AdjacencyList) MetricDependenciesTransitiveType {
-	depsCount := make(MetricDependenciesTransitiveType)
+/*
+Get transitive dependencies given adjacency list (used both for dependencies and dependents)
```

Comment:
Comments [should start ](https://go.dev/wiki/CodeReviewComments#comment-sentences)with a function name.